build/Targets/Settings.props

```diff
@@ -26,8 +26,10 @@
     <Configuration Condition="'$(Configuration)' == ''">Debug</Configuration>
     <SignAssembly>true</SignAssembly>
     <UseRoslynAnalyzers Condition="'$(UseRoslynAnalyzers)' == ''">true</UseRoslynAnalyzers>
-    <OutDir>$([System.IO.Path]::GetFullPath('$(MSBuildThisFileDirectory)\..\..\Binaries\$(Configuration)'))\</OutDir>
-    <BaseIntermediateOutputPath>$([System.IO.Path]::GetFullPath('$(MSBuildThisFileDirectory)\..\..\Binaries\Obj\$(MSBuildProjectName)'))\</BaseIntermediateOutputPath>
+    <BaseOutputPath Condition="'$(BaseOutputPath)' == ''">$(RepoRoot)Binaries\</BaseOutputPath>
+    <OutDir>$(BaseOutputPath)$(Configuration)\</OutDir>
+    <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)' == ''">$(RepoRoot)Binaries\Obj\</BaseIntermediateOutputPath>
```

Comment:
There is https://github.com/NuGet/Home/issues/4463  for project.assets.json collision. It seems from discussion we can't just rename/move that file without breaking things (linked project system bug).

We're repeating what common targets already do with BaseIntermediateOutputPath there (just need it to happen sooner). We *might* be able to append project file name to intermediate output path when it is not inside project dir. This might lead to duplicate project file names in path for users who have already set BaseIntermediateOutputPath to include it and move over to SDK. The design of BaseOutputPath in SDK follows the pre-existing BaseIntermediateOutputPath from common targets: up to you if you want to put project name in it or share in a different directory. I think it should stay like this.